src/python/pants/backend/experimental/jvm/goals/coursier_integration_test.py

```diff
@@ -0,0 +1,199 @@
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.experimental.jvm.goals.coursier import CoursierResolve
+from pants.backend.experimental.jvm.goals.coursier import rules as coursier_goal_rules
+from pants.backend.experimental.jvm.resolve.coursier_fetch import (
+    CoursierLockfileEntry,
+    CoursierResolvedLockfile,
+    MavenCoord,
+    MavenCoordinates,
+)
+from pants.backend.experimental.jvm.resolve.coursier_fetch import rules as coursier_fetch_rules
+from pants.backend.experimental.jvm.resolve.coursier_setup import rules as coursier_setup_rules
+from pants.backend.experimental.jvm.target_types import JvmDependencyLockfile
+from pants.backend.experimental.jvm.util_rules import rules as util_rules
+from pants.core.util_rules import config_files, source_files
+from pants.core.util_rules.external_tool import rules as external_tool_rules
+from pants.engine.fs import CreateDigest, Digest, FileContent, FileDigest
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        preserve_tmpdirs=True,
+        rules=[
+            *config_files.rules(),
+            *coursier_fetch_rules(),
+            *coursier_goal_rules(),
+            *coursier_setup_rules(),
+            *external_tool_rules(),
+            *source_files.rules(),
+            *util_rules(),
+        ],
+        target_types=[JvmDependencyLockfile],
+    )
+
+
+def get_digest(rule_runner: RuleRunner, source_files: dict[str, str]) -> Digest:
+    files = [FileContent(path, content.encode()) for path, content in source_files.items()]
+    return rule_runner.request(Digest, [CreateDigest(files)])
+
+
+def test_coursier_resolve_creates_missing_lockfile(rule_runner: RuleRunner) -> None:
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+        coursier_lockfile(
+            name = 'example-lockfile',
+            maven_requirements = [
+                'org.hamcrest:hamcrest-core:1.3',
+            ],
+        )
+        """
+        ),
+    )
+    result = rule_runner.run_goal_rule(CoursierResolve, args=["::"])
+    assert result.exit_code == 0
+    assert result.stderr == "Updated lockfile at: coursier_resolve.lockfile\n"
+    expected_lockfile = CoursierResolvedLockfile(
+        entries=(
+            CoursierLockfileEntry(
+                coord=MavenCoord(coord="org.hamcrest:hamcrest-core:1.3"),
+                file_name="hamcrest-core-1.3.jar",
+                direct_dependencies=MavenCoordinates([]),
+                dependencies=MavenCoordinates([]),
+                file_digest=FileDigest(
+                    fingerprint="66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9",
+                    serialized_bytes_length=45024,
+                ),
+            ),
+        )
+    )
+    expected_lockfile_path = os.path.join(rule_runner.build_root, "coursier_resolve.lockfile")
+    with open(expected_lockfile_path, "rb") as f:
```

Comment:
More modern is `assert Path(rule_runner.build_root, "coursier_resolve.lockfile").read_bytes() == ...`, from `pathlib` library.

Reply:
Done